model/DataObject.php

```diff
@@ -1647,6 +1648,121 @@ public function getRelationClass($relationName) {
 	}
 
 	/**
+	 * Given a relation name, determine the relation type
+	 *
+	 * @param string $component Name of component
+	 * @return string has_one, has_many, many_many, belongs_many_many or belongs_to
+	 */
+	public function getRelationType($component) {
+		$types = array('has_one', 'has_many', 'many_many', 'belongs_many_many', 'belongs_to');
+		foreach($types as $type) {
+			$relations = Config::inst()->get($this->class, $type);
+			if($relations && isset($relations[$component])) {
+				return $type;
+			}
+		}
+		return null;
+	}
+
+	/**
+	 * Given a relation declared on a remote class, mock the component.
```

Comment:
`Mock` as used in testing, i.g. I will get a component (or a components, it seems to be different from the @return) that are not the real objects, but behave like them?